### PR TITLE
Removed unused `encoder_hidden_states` and `encoder_attention_mask` 

### LINKED ISSUE #8969: MobileBERT decoder capabilities
The current input parameters for MobileBERT indicate that the model may be used in a decoder setting. However, the model architecture does not contain a cross-attention mechanism and several inputs to the model are effectively never used: `encoder_hidden_states` and `encoder_attention_mask`.

This can be seen in:
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L247, where these 2 inputs are not used
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L330, where these inputs are just passed to the previous forward function (where they have no impact)
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L496, where these parameters are not used (not even passed to the `MobileBertAttention`)
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L552 where they are passed to the `MobileBertLayer` described above (therefore without impact)
- https://github.com/huggingface/transformers/blob/de6befd41f3986c68f4af302761b627cb6519eb7/src/transformers/models/mobilebert/modeling_mobilebert.py#L847 where they will trigger some reshaping of the attention mask, but eventually not get used.

I believe these unused inputs make the code more difficult to follow and potentially misleading (I don't believe the model can actually be used as a decoder).

Would you be generally supportive of a cleanup of the MobileBERT architecture to reflect its current capabilities? I'd be happy to share a PR but I wanted to check your general thoughts on this.

Thank you,

### Who can help
 albert, bert, GPT2, XLM: @LysandreJik 
(did not find anyone for MobileBert? But this is relevant for package maintenance I believe you may be the right person for this)

### PR DESCRIPTION
**Pull request by guillaume-be**
# What does this PR do?

This PR removes unused `encoder_hidden_states` and `encoder_attention_mask` from MobileBert forward methods. These are use for decoder models, but MobileBert does not include a cross-attention mechanism.

Fixes https://github.com/huggingface/transformers/issues/8969

## Who can review?
 albert, bert, XLM: @LysandreJik

### CODE REVIEW

**Comment by guillaume-be** [comment]
@LysandreJik I had to remove some tests that were testing the decoder mode for MobileBert. 

One test still fails (flax), the error seems unrelated to this PR unless I am missing something?

**Review by LysandreJik — APPROVED** [review]
This LGTM, it doesn't remove any supported behavior. @patrickvonplaten what do you think?

**Review by patrickvonplaten — APPROVED** [review]
Thanks a lot for cleaning that @guillaume-be
